parcels/field.py

```diff
@@ -967,15 +968,10 @@ def __init__(self, name, U, V, W=None):
 
     def dist(self, lon1, lon2, lat1, lat2, mesh):
         if mesh == 'spherical':
-            r = 360*60*1852/2/np.pi
             rad = np.pi/180.
-            x1 = r*np.cos(rad*lon1) * np.cos(rad*lat1)
-            y1 = r*np.sin(rad*lon1) * np.cos(rad*lat1)
-            z1 = r*np.sin(rad*lat1)
-            x2 = r*np.cos(rad*lon2) * np.cos(rad*lat2)
-            y2 = r*np.sin(rad*lon2) * np.cos(rad*lat2)
-            z2 = r*np.sin(rad*lat2)
-            return np.sqrt((x2-x1)**2 + (y2-y1)**2 + (z2-z1)**2)
+            deg2m = 1852 * 60.
+            lat = (lat1+lat2)/2.
+            return np.sqrt(((lon2-lon1)*deg2m*cos(rad * lat))**2 + ((lat2-lat1)*deg2m)**2)
```

Comment:
So we are using the assumption that distances are simple L2 norm on the plane tangental to the sphere? That only works for small distances between points. Why make this change then, if the former method was more general?

Reply:
This change is more complicate.
Basically, we pass twice from deg/s to m/s: the first time here when we compute the fluxes; the second time in the interpolation while we divide by the jacobian, which takes into account the changes from curvi to parent element + the spheric mesh.
This new change is more consistent because we use the same transformation twice, while before the transformation where independent one from each other.

This way is also consistent with what we do for 3D interpolation. And in 3D, it is the only way to do it.

As you said, it is less accurate (theoretically). It is true, but only because what I do is still currently wrong. For the moment I evaluate at the lat as the mean lat of the edge. If I take the correct one (not the particle lat, but the position as it is done in 3D (see fig. 5 of the draft), it will be consistent and accurate). For the moment, the two transformations we do are not consistent.

Practically, I observed that it does not change anything.
I'll check with more details to see if it is a bit more accurate with the new scheme (while corrected).